### DISCUSSION TITLE
Moving from Z-Stick Gen5 to Z-stick 7

### DISCUSSION THREAD

**Discussion by Micka41** [zwave-js/zwave-js#2927] [Q&A] [answered]
Hi,

How can I moving from Z-Stick Gen5 to Z-stick 7 (about 90 devices) ?

Thanks

**Comment by robertsLando** [comment]
AFAIK the only way is to removve all the devices from the first stick and add them back one by one to the other

**Reply by Micka41** [reply]
Ah ah ah, your are funny !!

**Reply by AlCalzone** [reply]
Z-Wave has the possibility to add a second controller to the network, then transfer it the primary role. Unfortunately this is not yet supported by zwave-js, maybe you can do that with Silabs' PC Controller software.

**Reply by Micka41** [reply]
I didn't find any transfert/duplicate functionality in the Silabs documentation.
I think there will be "many" people asking for this transfert functionality.

**Reply by kpine** [reply]
[PC Controller User's Guide](https://www.silabs.com/documents/public/user-guides/INS13114-Z-Wave-PC-Based-Controller-v5-User-Guide.pdf), Section 4.3.3 Controller Shift.

[Aeotec Guide](https://help.aeotec.com/support/solutions/articles/6000129218-perform-a-controller-shift-to-change-primary-roles-using-2-z-sticks-), using an old version of the PC Controller software.

Good luck! (I've never done it)

**Reply by drobtravels** [reply]
So... anyone attempted the above?

I just tried it, using the SI Labs PC Controller to move to the Zooz 700 controller. 

I was able to successfully do the initial controller shift to the Zooz 700.  However, now I can't seem to shift the primary *back* to my old zwave stick so that I can change the node ID.  Right now the Zooz 700 is primary with a node id of 98 (should be 1).  I'm able to control devices from the PC controller.. but assume this isn't going to work once I move back to Zwave JS.  

Anyone have success or advice?

**Reply by AlCalzone** [reply]
> advice

If you can wait a couple of days (and have an NVM backup), I'm working on a tool to make this easier.
When its done, the process will be:
- make NVM backup of 500 series stick
- switch to 700 series stick
- upgrade NVM backup with the tool
- restore upgraded backup on 700 series stick
- done

**Reply by drobtravels** [reply]
I think my zwave stick is too old to support NVM backup... Although ZwaveJS did spit out a file..

Anyways I can't seem to make the 500 primary again. Any way to use in zwave js when controller node id is not 1, or do I need to recreate my network?

**Reply by drobtravels** [reply]
Nevermind.. was able to do the NVM restore in ZwaveJS (despite PC controller saying its not supported) and am back up and running on original 500 series stick.  Will wait until you release your new tool to proceed.  Thanks for the heads up!

**Reply by AlCalzone** [reply]
> Any way to use in zwave js when controller node id is not 1, or do I need to recreate my network?

It doesn't care about the node id. Only about being the primary controller.

**Reply by GrizzlyAK** [reply]
> If you can wait a couple of days (and have an NVM backup), I'm working on a tool to make this easier. 

Hey, @AlCalzone, are you still looking for backup files to test? I seem to recall you were looking for some, but my stick had the old architecture and wouldn't work. I tried to update my FW on my original Z-Stick Gen5 updated to work with the Pi on USB3, but never could. I reached out to Chris at Aeotec and he sent me a new 5+ with the new FW already installed, and I just loaded my config on it and have it up and running. If you still need samples, let me know and I can send you one.

**Reply by AlCalzone** [reply]
The migration is now included in zwavejs2mqtt: https://github.com/zwave-js/node-zwave-js/issues/3906#issuecomment-997484466
There still seems to be something strange when moving from 500 -> 700 series, but I haven't had the time to look into it yet. I've got backup files though.

**Comment by Hedda** [comment]
> Z-Wave has the possibility to add a second controller to the network, then transfer it the primary role. Unfortunately this is not yet supported by zwave-js, maybe you can do that with Silabs' PC Controller software.

@AlCalzone Have you given any further thoughts on adding support for a secondary controller to Z-Wave JS?

That is to among other things add the possibility to migrate without using the existing NVM backup and NVM restore method.

Personally, I would however love to always have a secondary controller installed and online for redundancy.

**Comment by niemyjski** [comment]
Any updates on this?

**Reply by AlCalzone** [reply]
What kind of update are you looking for?

**Reply by Hedda** [reply]
Just a guessing but think that he might is looking for a step-by-step guide and/or automated process for cross-hardware-generation backup and restore procedure in order to achieve a migration from a Z-Wave 500 series controller to a Z-Wave 700/800 series controller?

**Reply by robertsLando** [reply]
Just make an nvm backup of the gen5 controller and restore that to the 700 series controller

**Reply by niemyjski** [reply]
Yes, I was looking for a way to do this via UI or official documentation. An automated process or having a button to backup and restore in the ui would be amazing (I use HA)... I have the original aeotec gen5 (there are multiple revisions) so I wasn't sure if I could even do a migration as I don't have even firmware updates available via their website (the later revisions have firmware updates). But I can do an offline backup but it's in binary format.

Is an nvm backup possible? If so how?

**Reply by robertsLando** [reply]
I speak about zwave-js UI, in order to do an nvm backup/restore just go to control panel > advanced actions > nvm backup/restore

**Reply by Hedda** [reply]
> Yes, I was looking for a way to do this via UI or official documentation. An automated process or having a button to backup and restore in the ui would be amazing (I use HA).

I agree that would be a very nice feature if a such feature existed and is maybe even something that Home Assistant users will come to expect now that Home Assistant has implemented such features for Zigbee.


Off-topic but FYI; [Home Assistant](https://www.home-assistant.io) can do migration from Zigbee2MQTT to [ZHA](https://www.home-assistant.io/integrations/zha) + migration from old adapter in [ZHA](https://www.home-assistant.io/integrations/zha) to new adapter in [ZHA](https://www.home-assistant.io/integrations/zha):

https://skyconnect.home-assistant.io/migrate-zigbee-zha/ (Guide on how to migrate from a Zigbee2MQTT network to [ZHA](https://www.home-assistant.io/integrations/zha) using [ZHA](https://www.home-assistant.io/integrations/zha)).

https://skyconnect.home-assistant.io/migrate-zigbee-backup/ (Guide on how to migrate a backup from Zigbee2MQTT to [ZHA](https://www.home-assistant.io/integrations/zha) using [ZHA](https://www.home-assistant.io/integrations/zha)).

https://skyconnect.home-assistant.io/migrate-zigbee-zha-only/ (Guide on how to migrate old adapter to new adapter if already using [ZHA](https://www.home-assistant.io/integrations/zha)).

Note! Those guides were specifically written for Home Assistant SkyConnect (ezsp) adapter and are for migration to/inside [ZHA integration](https://www.home-assistant.io/integrations/zha). 

Might inspire other ideas but those guides are for example not meant to be used for migrating from [ZHA](https://www.home-assistant.io/integrations/zha) to Zigbee2MQTT, nor migrating between different adapters if continuing to use Zigbee2MQTT (e.g. not for migrating old to new Zigbee adapter if using Zigbee2MQTT).


PS: Again, Home Assistant's own [ZHA integration](https://www.home-assistant.io/integrations/zha) depends for backup and restore command from [zigpy/zigpy-cli](https://github.com/zigpy/zigpy-cli) to migrate Zigbee network, however both that [ZHA integration](https://www.home-assistant.io/integrations/zha) and Zigbee2MQTT use the same "[Open ZigBee Coordinator Backup Format](https://github.com/zigpy/open-coordinator-backup)" standard for its backup files.

**Reply by GrizzlyAK** [reply]
> Just make an nvm backup of the gen5 controller and restore that to the 700 series controller

Hi Daniel, is this a reality now? Has anyone done this with the Aeotec sticks themselves (500 -> 700)? I'm still on a 5+and would love to upgrade to 700 if it is a seamless process. It sounded like @AlCalzone was working on a tool, but some problems existed with 500->700. I don't suspect it is an easy problem to solve, especially given the various controllers. I loathe the thought of rebuilding my network from scratch. Cheers.

**Reply by AlCalzone** [reply]
Yeah, it's been for over a year now:
https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG_v8.md#890-2021-12-22

The initial version saw some revisions though, but most quirks have been solved since June 22: https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md#940-2022-06-10

For the Z-Stick 5(+), you need it to be on firmware 1.2 though, otherwise Z-Wave JS doesn't understand the NVM format.

**Comment by GrizzlyAK** [comment]
THIS is outstanding! I'll see if I can figure out what version my FW is. I'm pretty sure I've done an NVM backup before after I got my 5+ replacement. Thanks again for all the work you and Daniel do to make ZWave _**the**_ place to be. This changelog is a testament to the amount of effort put in.

BTW, I'm looking for an alternative to my Paypal sponsorship as that is where I consolidate my payments for many things. GitHub changing this was a really BAD move for everyone (but them). Cheers.

**Reply by niemyjski** [reply]
If you are using only ha, what's the easiest process to do the migration?

**Reply by kpine** [reply]
https://www.home-assistant.io/integrations/zwave_js/#how-do-i-switch-between-the-official-z-wave-js-add-on-and-the-z-wave-js-ui-add-on

**Reply by robertsLando** [reply]
Thanks @GrizzlyAK ❤️ About GH paypal move I agree with you and sincerly I dunno why this happened.

BTW Patreon still supports PayPal: https://patreon.com/user?u=16906849&utm_medium=clipboard_copy&utm_source=copyLink&utm_campaign=creatorshare_creator&utm_content=join_link

**Comment by GrizzlyAK** [comment]
@robertsLando I sent you an email re: Patreon.

**Comment by AlCalzone** [comment, marked as answer]
Commenting again, so I can mark this as an answer:

1. Make sure the Z-Stick Gen 5 is on [firmware 1.2](https://aeotec.freshdesk.com/support/solutions/articles/6000252294-z-stick-gen5-v1-02-firmware-update). You may need a Windows machine for this.
2. Perform an NVM backup using Z-Wave JS UI
3. Plug the Z-Stick 7 in
4. Restore the previous NVM backup

The opposite direction is also possible.

**Reply by AlCalzone** [reply]
Try restarting Z-Wave JS UI or reloading the page.

**Reply by GuiPoM** [reply]
Thank you. A restart solved that minor issue :)

**Reply by breversa** [reply]
> The opposite direction is also possible.

Can someone confirm?

I'm about to migrate (back) from a Z-Stick 7 to a Z-Stick 5+ and I don't want to brick anything. :-(

**Comment by niemyjski** [comment]
@AlCalzone what about First Generation Z-Stick Gen 5 which never got a firmware update? These were the popular ones on amazon (I have 2)

**Reply by Hedda** [reply]
> what about First Generation Z-Stick Gen 5 which never got a firmware update? These were the popular ones on amazon

Not tested myself but can flash it with newer unofficial firmware if can find out exactly which Z-Wave device chip is on the board?

https://aeotec.freshdesk.com/support/solutions/articles/6000167090-check-product-firmware-using-zensys-tool

Here is for example a user who flashed Aeon Labs Z-Stick Gen5 ZW090 with new firmware update released for Aeotec Z-Stick Gen5+:

https://community.home-assistant.io/t/aeotec-z-stick-gen5-zw090-new-firmware-update/357972

https://aeotec.freshdesk.com/support/solutions/articles/6000252294-z-stick-gen5-v1-02-firmware-update

**Reply by AlCalzone** [reply]
@niemyjski the firmware I linked **IS** for the Gen5. It will update it to the same firmware that the Gen5+ uses.
Or am I misunderstanding your question?

**Reply by niemyjski** [reply]
@AlCalzone No there is no firmware update. Here is a snippet of an email chain with aeotec support:

**My response**

> I couldn't get either to work but I think I know the problem. In 2018 and 2019 I bought the gen5 stick. Both backs look like this and so I don't think they are the gen5+ :\... It was the most popular zwave stick still on amazon :\ Does this have a firmware update and what is the difference to the gen5+?

![image](https://user-images.githubusercontent.com/1020579/224043484-60209c00-475c-4ab7-a83d-5e85a3ab1b6c.png)

**Their response**

> Unfortunately, there is no way to update from a Z-Stick Gen5 to a Z-Stick 7 in any case as the memory format is too different so the transfer of networks becomes impossible. - Only method is to unpair and re-pair devices unfortunately moving onto series 700. 
​
> It is an older generation Z-Stick Gen5 based on the plastics on the device itself, but the firmware update and SDK differences are negligible for typical use and you probably won't notice too many differences between the versions if at all any. 

**My response**

> Ok, thanks for the info. Is it possible to get a firmware update for the gen5 with the fixes of the gen5+

**Their response**

> Gen+ and New Z-Stick Gen5+ are hardware changes that allow the connection to the USB chipsets used which is the primary difference, firmware update themselves won't resolve the issue. I can advise how to modify the hardware if you are seeing issues with connection to RPi4 and if you are comfortable with resoldering a resistor on the Z-Stick Gen5.

**Reply by Hedda** [reply]
Sounds like they misunderstood your question or you were unclear with them which confused them, (you should really not have mentioned Z-Stick 7 as from their point of view that has nothing to do with what you should be asking about). As you do not want to upgrade Aeotec Z-Stick Gen5 ZW090 with firmware made for Z-Stick 7 (which it sounds like they are replying to). Instead, clarify to them that all you want to do is to flash your Aeotec Z-Stick Gen5 ZW090 stick with "z-stick gen5 v1-02 firmware". What you do after you updated your Aeotec Z-Stick Gen5 ZW090 stick with "z-stick gen5 v1-02 firmware" has nothing directly to do with them. That is, the migration that done with zwave-js is nothing that Aeotec official support themselves. 

https://aeotec.freshdesk.com/support/solutions/articles/6000252294-z-stick-gen5-v1-02-firmware-update

https://community.home-assistant.io/t/aeotec-z-stick-gen5-zw090-new-firmware-update/357972

**Reply by AlCalzone** [reply]
@niemyjski you're mixing up a couple of things.

1. Gen5 and Gen5+ are different in that they have a hardware modification. Gen5 is violating the USB standard and needs a powered USB hub to work with a Raspberry Pi 4. Gen5+ is fine in this regard.
2. Gen5 and Gen5+ originally had different firmwares. Gen5 was on 1.0 I think, Gen5+ had an updated firmware v1.2
3. The link I posted you above does contain a firmware update for the Gen5 (non +) to firmware v1.2:
![image](https://user-images.githubusercontent.com/17641229/224053339-2a9472a9-0604-4734-8aa5-2f78ac4e7106.png)
Unfortunately this page mentions Gen5+ pretty prominently, so a lot of people are confused.
4. And last but not least, your stick **must** be on firmware 1.2, so Z-Wave JS can migrate its NVM to the format the 700 series uses. I don't think that the Aeotec support knows about this method.

**Reply by niemyjski** [reply]
Thanks for the clarification! Yeah, I was asking them about newer firmware and how to upgrade to 7 series. This convo was from last year.

**Reply by Hedda** [reply]
Yeah as far as Aeotec support will be aware there are no seamless/lossless migration paths that they provide in a first-party solution. Obviously, the zwave-js migration path method is an unofficial third-party method from Aeotec's point of view. Do not expect end-to-end help or knowledge from Aeotec support on how to migration from a Z-Wave 500 stick to a Z-Wave 700 stick or Z-Wave 800 stick.

**Comment by beno57** [comment]
Hi everyone,

I've a problem with my 700 zooz to restore NVM.
I've a gen5 and I updated the firmware with aeotec tools. Now I've a 1.2 FW on my gen5

I backuped a NVM on my gen5 but when I import the NVM on my gen7, I received an error with NVM convert

Error while calling restoreNVM: Invalid NVM JSON: Controller is part of another network but has no learned Home ID (ZW0281)

zwave-js-ui: 8.19.0.487524b
zwave-js: 11.0.0

Do you have an idea why I can't import the nvm file on my gen7 ?

many thx

**Reply by AlCalzone** [reply]
Fixed in yesterday's release: https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md#1170-2023-07-27

**Reply by beno57** [reply]
Nice, thx for your reply :)

**Comment by HFeenstra** [comment]
For those struggling with an Aeotec Gen 5 stick, 1.0 version to a 1.2(1.02) version to get a backup for a higher 700/800 version stick, follow my steps in the last comment in this thread:

https://community.home-assistant.io/t/how-to-best-migrate-from-aeotec-gen-5-to-aeotec-z-stick-7/637311/18?u=hfeenstra

**Comment by soso-bc** [comment]
Hi,

I have an old  [Everspring EVR_SA413-1 Zwave+ 500 series stick](https://www.domadoo.fr/fr/interface-domotique/3619-everspring-controleur-usb-z-wave-4713616113625.html).

It is recognized as this by ZwaveJS :

```
"productLabel": "HUSBZB-1",
"productDescription": "QuickStick Combo",
"manufacturer": "Sigma Designs (Former Zensys)",
"firmwareVersion": "4.36",
"sdkVersion": "6.51.7",
 "protocolVersion": 3,
```

I'd like to upgrade to an 800 series stick because I can no longer integrate the latest generation devices.

Is migration via NVM backup/restore possible to a Gen 8 stick?

If not, to a Gen 7?

A related question: will communication between my Gen 5/Gen 8 devices be compromised (not a compatibility issue)?

Thank's

**Reply by AlCalzone** [reply]
It's complicated:

https://community.hubitat.com/t/guide-nortek-husbzb-1-nvm-backup-restore-and-updating-z-wave-firmware/48012

> will communication between my Gen 5/Gen 8 devices be compromised (not a compatibility issue)?

It should not, unless your new stick is worse than the old one.

**Reply by soso-bc** [reply]
I may not have understood everything in the link:
- I don't want to update the firmware of the key (I don't have any operating issues)
- My key is slightly more up-to-date (FW 4.36 vs 4.05/SDK 6.51.7 vs 6.51.06) - I bought it in 2017 (the tutorial mentions SDKs from 2014/2015)
- It seems that backup/restore already worked in FW 4.05 / SDK 6.51.06

But my question was rather: is restoring the backup (gen5) to a newer-generation chip possible/supported (gen8/7)?

I think to be on the safe side, I'll leave my old key and mount a new Docker instance of zwavejs-ui with the new key...

**Reply by AlCalzone** [reply]
To be able to have Z-Wave JS convert the backup to a format compatible with 700/800 series, the SDK of the source needs to be at least 6.61, so you will have to update the firmware to go that route.

Directly restoring the backup without that conversion is not possible.

**Reply by soso-bc** [reply]
Okay, I hadn't noticed that :/
I'm going to set up a second zwavejs-ui instance; it'll be easier ;)
I just hope there won't be any interference between the two sticks. To be sure, I'll put the two sticks on the ends of USB extension cables to keep them apart.

Thank's a lot !

**Comment by mazmar** [comment]
Sorry, I first posted this in wrong github (zwavejs-ui) but i have issue migrating from Aeotec ZStick5 to ZStick 10Pro

I tried doing backup restore but after restore everything is asleep, and doesn't communicate. From logs I don't see any error or something

versions:

```
zwave-js-ui: 11.1.0.c235c6c
zwave-js: 15.10.0
home id: 3603720926
home hex: 0xd6cc6ade
```

ZStick 5 FW 1.2, SDK: v6.81.6
ZStick 10Pro FW 2.0 SDK: V7.23.2

 
[restore_log_debug.txt](https://github.com/user-attachments/files/21815921/restore_log_debug.txt)

**Reply by AlCalzone** [reply]
There seems to be something wrong with the RF configuration on the new controller.
I don't find the issue right now where this was discussed before, but it is probably related to a firmware bug.

If you send me your NVM backup of the stick after restoring, I can try to fix it for you - no guarantees though.

**Reply by mazmar** [reply]
Here,

[NVM_20250818104822.bin.zip](https://github.com/user-attachments/files/21833506/NVM_20250818104822.bin.zip)

It's brand new Z-Stick 10Pro

**Reply by AlCalzone** [reply]
Ok, I need to check something else to see if this is still the aforementioned firmware bug, or something in the migration routine of Z-Wave JS. Can you factory reset (hard reset) the Z-Stick 10, and afterwards make another NVM backup and send me that?

Also please send me the backup from the Z-Stick 5 you wanted to migrate from.

**Reply by mazmar** [reply]
[NVM_20250818111543—z10-after-reset.bin.zip](https://github.com/user-attachments/files/21833958/NVM_20250818111543.z10-after-reset.bin.zip)
[NVM_20250816182452—restoration.bin.zip](https://github.com/user-attachments/files/21833961/NVM_20250816182452.restoration.bin.zip)

**Reply by mazmar** [reply]
@AlCalzone any news on this? Or should I send it back for warranty?

**Reply by AlCalzone** [reply]
Sorry for the delay. This is a problem with how custom firmware versions (not aligned with the SDK version) interfere with the internal NVM upgrade procedure of the Z-Wave firmware.

I **think** they are now working around it by using the "standard" firmware versions, at least it looks like this on the page about firmware updates:
https://aeotec.freshdesk.com/support/solutions/articles/6000279576-z-stick-10-pro-update-z-wave-v7-23-2-with-windows

You could try updating and then retrying the migration.
